tests/networkpolicy_test.go

```diff
@@ -9,6 +9,8 @@ import (
 	v1network "k8s.io/api/networking/v1"
```

Comment:
intended for commit `tests, Add waitForNetworkPolicyDeletion` :

Does this commit intend to fix flakiness? 
If so, please state it on the commit desc.
if not, then I'm not sure why it is relevant to "making networkpolicy dual-stack compatible".
can you please elaborate?

Reply:
it can fix flakiness (not the only one that was here) because the nature of the policies, 
each group of test there use a common policy and expect only this policy to exists, we dont want leftovers from previous tests that might cause an unexpected behavior.
each `Context` is based only on the policy its create and assumes only this policy exists.

As part of the effort, those fixes were applied, i might move the last commit to other PR as Edy suggested, and it would be less confusing.
its not about dual stack, it was a fix to make the tests better and more stable.

Reply:
ok but if you keep this commit in the PR then please mention in the desc. that this commit means to fix flakiness. 
will make it easier for reviewer to have the proper context

Reply:
Done